pkg/pillar/cmd/zedagent/parseconfig.go

```diff
@@ -763,39 +759,35 @@ func parseOneSystemAdapterConfig(getconfigCtx *getconfigContext,
 		switch port.Dhcp {
 		case types.DT_STATIC:
 			if port.AddrSubnet == "" {
-				errStr := fmt.Sprintf("parseSystemAdapterConfig: Port %s DT_STATIC but missing "+
+				err := fmt.Errorf("parseSystemAdapterConfig: Port %s DT_STATIC but missing "+
 					"subnet address in %+v; ignored", port.IfName, port)
-				log.Error(errStr)
-				port.ParseError = errStr
-				port.ParseErrorTime = time.Now()
+				log.Error(err)
+				port.SetError(err, "zedagent parseconfig")
 			}
 		case types.DT_CLIENT:
 			// Do nothing
 		case types.DT_NONE:
 			if isMgmt {
-				errStr := fmt.Sprintf("parseSystemAdapterConfig: Port %s: isMgmt with DT_NONE not supported",
+				err := fmt.Errorf("parseSystemAdapterConfig: Port %s: isMgmt with DT_NONE not supported",
 					port.IfName)
-				log.Error(errStr)
-				port.ParseError = errStr
-				port.ParseErrorTime = time.Now()
+				log.Error(err)
+				port.SetError(err, "zedagent parseconfig")
 			}
 		default:
-			errStr := fmt.Sprintf("parseSystemAdapterConfig: Port %s: ignore unsupported dhcp type %v",
+			err := fmt.Errorf("parseSystemAdapterConfig: Port %s: ignore unsupported dhcp type %v",
 				port.IfName, network.Dhcp)
-			log.Error(errStr)
-			port.ParseError = errStr
-			port.ParseErrorTime = time.Now()
+			log.Error(err)
+			port.SetError(err, "zedagent parseconfig")
 		}
 		// XXX use DnsNameToIpList?
 		if network != nil && network.Proxy != nil {
 			port.ProxyConfig = *network.Proxy
 		}
 	} else if isMgmt {
-		errStr := fmt.Sprintf("parseSystemAdapterConfig: Port %s isMgmt without networkUUID not supported",
+		err := fmt.Errorf("parseSystemAdapterConfig: Port %s isMgmt without networkUUID not supported",
 			port.IfName)
-		log.Error(errStr)
-		port.ParseError = errStr
-		port.ParseErrorTime = time.Now()
+		log.Error(err)
+		port.SetError(err, "zedagent parseconfig")
```

Comment:
These errors will be user-visible, hence should be actionally. I'd argue that the old message string "parseSystemAdapterConfig:"  doesn't help explain the issue and how the user should resolve it. But adding a "zedagent parseconfig" to the string makes it worse.

Reply:
Yeah.. Agree.. Filled it from the developer perspective..

Currently - 2 places we are getting the error from:

1) Parse Config -- I will put the source as - "Config Sanity Check"
2) Error from testing the config - Source: "Network Test"